src/plugins/workspace/public/plugin.ts

```diff
@@ -33,9 +43,60 @@ export class WorkspacePlugin implements Plugin<{}, {}, {}> {
     const workspaceId = this.getWorkspaceIdFromURL(core.http.basePath.getBasePath());
 
     if (workspaceId) {
-      core.workspaces.currentWorkspaceId$.next(workspaceId);
+      const result = await workspaceClient.enterWorkspace(workspaceId);
+      if (!result.success) {
+        /**
+         * Fatal error service does not support customized actions
+         * So we have to use a self-hosted page to show the errors and redirect.
+         */
+        (async () => {
+          const [{ application, chrome }] = await core.getStartServices();
+          chrome.setIsVisible(false);
+          application.navigateToApp(WORKSPACE_FATAL_ERROR_APP_ID, {
+            replace: true,
+            state: {
+              error: result.error,
```

Comment:
do we consider to to check result.error exists before access it. like as `const error = result.error || "Unknown error";` so ensures that `error` always have value

Reply:
If the error is empty or null, we will hide the callout component to not display the detailed error message, but update to `result?.error` to protect the code.